flang/lib/Parser/prescan.cpp

```diff
@@ -597,6 +595,30 @@ const char *Prescanner::SkipWhiteSpace(const char *p) {
   return p;
 }
 
+const char *Prescanner::SkipWhiteSpaceIncludingEmptyMacros(
+    const char *p) const {
+  while (true) {
+    if (int n{IsSpaceOrTab(p)}) {
+      p += n;
+    } else if (preprocessor_.AnyDefinitions() && IsLegalIdentifierStart(*p)) {
+      // Skip keyword macros with empty definitions
+      const char *q{p + 1};
+      while (IsLegalInIdentifier(*q)) {
```

Comment:
Can't identifiers contain numbers but not start with them? Will this code hold up to that?

Reply:
Yes, and yes.  The names `IsLegalIdentifierStart` and `IsLegalInIdentifier` should be clear enough.